rioxarray/rioxarray.py

```diff
@@ -230,7 +216,79 @@ def crs(self):
         """:obj:`rasterio.crs.CRS`:
             The projection of the dataset.
         """
-        return self._crs
+        raise NotImplementedError
+
+    def set_crs(self, input_crs):
+        """
+        Set the CRS value for the Dataset/DataArray without modifying
+        the dataset/data array.
+
+        Parameters:
+        -----------
+        input_crs: object
+            Anythong accepted by `rasterio.crs.CRS.from_user_input`.
+        """
+        self._crs = CRS.from_user_input(input_crs)
+
+    def write_crs(
+        self, input_crs=None, grid_mapping_name=DEFAULT_GRID_MAP, inplace=False
+    ):
+        """
+        Write the CRS to the dataset in a CF compliant manner.
```

Comment:
Is there a reason CF is the only available option for such a generic method name (`write_crs`)? Do any of the existing methods assign the CRS object to a `crs` coordinate?

Reply:
> Is there a reason CF is the only available option for such a generic method name

It is currently the only method supported by `rioxarray` for writing the CRS to the dataset. Did you have other conventions you were thinking of following for writing?

> Do any of the existing methods assign the CRS object to a crs coordinate?

You can assign it to the CRS coordinate using the `grid_mapping_name`. You can see an example of this in the tests in the PR.

Reply:
> Did you have other conventions you were thinking of following for writing?

No, you're right. CF is really the only "new" format that needs conversion I guess. Except maybe pyproj CRS -> rasterio CRS. I guess I was just mainly noticing that this method writes to the xarray object associated with it and I could see that being useful for things like setting a `crs` coordinate or adding `y` and `x` coordinates. Likely something more in geoxarray's "generic" realm though.

Reply:
Ah, you want to attach the `CRS` object to the coordinate instead of the CF parameters to the coordinate. No, that is currently not supported.

Reply:
Yes, that was my hope for geoxarray (`data_arr.coords['crs'].item()` is a pyproj.crs.CRS object). Maybe I'll finally get the time to do it during SciPy...yeah right.

Reply:
> data_arr.coords['crs'].item() is a pyproj.crs.CRS object

Just curious, what is the benefit of that versus having the property `geo.crs`? I guess it does come into play with checking the equality of two DataArrays. Any other reasons? My only concern with that would be that it would make it so the dataset is no longer serializable and cannot be written with `to_netcdf`. But. I guess you could add `geo.to_netcdf`.
(maybe this is a discussion that should be moved to a separate issue in geoxarray?).

> Maybe I'll finally get the time to do it during SciPy

Well, you can always hope :crossed_fingers: 

> Except maybe pyproj CRS -> rasterio CRS

I think this is something that should be handled in `set_crs` (which will be called by `write_crs`).

Reply:
What determines the serialization of this? I suppose it would have to be handled in core xarray. Yes, this isn't a great situation. The goal was to have xarray treat the CRS as a "real" coordinate, but I later found out that xarray doesn't actually care if the CRS matches or not if it isn't a dimensional coordinate (like `y` or `x`) and will end up just dropping the `crs` coordinate if needed.

I wasn't a huge fan of `data_arr.geo.crs` because it felt "hidden" and seemed like it would be difficult to explain why this valuable information has to be accessed this way (thinking from a tutorial/teaching point of view).

Reply:
Here is the serialization issue I was talking about:
```python
>>> import xarray
>>> from rasterio.crs import CRS
>>> xds = xarray.Dataset()
>>> xds.coords["crs"] = xarray.Variable((), CRS.from_epsg(4326))
>>> xds
<xarray.Dataset>
Dimensions:  ()
Coordinates:
    crs      object <bound method Mapping.values of CRS.from_epsg(4326)>
Data variables:
    *empty*
>>> xds.to_netcdf("test_crs.nc")
...
ValueError: unable to infer dtype on variable 'crs'; xarray cannot serialize arbitrary Python objects
```

> I wasn't a huge fan of data_arr.geo.crs because it felt "hidden"  ...

I guess it is not really visible on the xarray dataset. I think if there is documentation and examples of proper usage, it has potential to be an okay path forward.

Reply:
I am going to mark this as resolved as the conversation has moved to something that should be in another issue.